src/plugins/intel_gpu/src/graph/include/primitive_type_base.h

```diff
@@ -58,7 +61,16 @@ struct primitive_type_base : primitive_type {
 
     bool does_possible_implementation_exist(const cldnn::program_node& node, const kernel_impl_params& impl_param) const override {
         OPENVINO_ASSERT(node.type() == this, "[GPU] primitive_type_base::does_possible_implementation_exist: primitive type mismatch");
-        return implementation_map<PType>::check_io_eq(impl_param, node.get_preferred_impl_type());
+        return implementation_map<PType>::check_io_eq(impl_param, node.get_preferred_impl_type(), shape_types::static_shape);
+    }
+
+    bool does_dynamic_implementation_exist(const cldnn::program_node& node) const override {
+        return does_dynamic_implementation_exist(node, *node.get_kernel_impl_params());
```

Comment:
Can't this be just called from does_possible_implementation_exist implicitly?

Reply:
Using does_possible_implementation_exist with shape type detection leads to bunch of errors for dynamic models. It seems that reorder_inputs pass in that case tries to do something weird. I didn't want to touch it at the moment, as it may impact static flow, thus I separated dynamic impl existence check into new method.

Reply:
Got it, thanks.